graphix/clifford.py

```diff
@@ -96,33 +119,31 @@ def commute_domains(self, domains: Domains) -> Domains:
         Note that applying the method to `self.conj` computes the reverse commutation:
         indeed, `C†X^sZ^t = (X^sZ^tC)† = (CX^s'Z^t')† = X^s'Z^t'C†`.
         """
-        s_domain = domains.s_domain
-        t_domain = domains.t_domain
+        s_domain = domains.s_domain.copy()
+        t_domain = domains.t_domain.copy()
         for gate in self.hsz:
-            if gate == graphix.clifford.I:
+            if gate == Clifford.I:
                 pass
-            elif gate == graphix.clifford.H:
+            elif gate == Clifford.H:
                 t_domain, s_domain = s_domain, t_domain
-            elif gate == graphix.clifford.S:
+            elif gate == Clifford.S:
                 t_domain ^= s_domain
-            elif gate == graphix.clifford.Z:
+            elif gate == Clifford.Z:
                 pass
             else:
                 raise RuntimeError(f"{gate} should be either I, H, S or Z.")
         return Domains(s_domain, t_domain)
 
-
-TABLE = tuple(Clifford(i) for i in range(len(CLIFFORD)))
-
-
-def get(index: int) -> Clifford:
-    """Return the Clifford gate with given index."""
-    return TABLE[index]
+    @staticmethod
+    def cliffords() -> Iterator[Clifford]:
+        """Return an iterator over the Clifford gates."""
+        for i in range(len(CLIFFORD)):
+            yield Clifford(i)
```

Comment:
It would be more idiomatic to be able to write `list(Clifford)` (as we would be able to do if `Clifford` was defined as an `Enum`).